### PR TITLE
Make Service Plan configurable for AZ Service Functions

### PR DESCRIPTION
**Pull request by goettl79**
##### SUMMARY

Make azure app plan configurable for azure functions

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

azure_rm_functionapp

##### ADDITIONAL INFORMATION

Currently, Azure functions can't be assigned a web app plan during creation. With this change, you can now use other than pay as you go web app plans.

### CODE REVIEW

**Comment by Fred-sun** [comment]
@goettl79 Thanks for your contribution, When you finished changed, Please let me know, I will push for review and merge. Thanks!

File: lib/ansible/modules/cloud/azure/azure_rm_functionapp.py [hunk 1]
```diff
@@ -34,6 +34,11 @@
     location:
         description:
             - Valid Azure location. Defaults to location of the resource group.
+    plan: 
+        description:
+            - "It can be resource id of existing app service plan. eg.,
+              /subscriptions/<subs_id>/resourceGroups/<resource_group>/providers/Microsoft.Web/serverFarms/<plan_name>"
+        required: false
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_functionapp.py]: required: false could be removed

File: lib/ansible/modules/cloud/azure/azure_rm_functionapp.py [hunk 2]
```diff
@@ -34,6 +34,11 @@
     location:
         description:
             - Valid Azure location. Defaults to location of the resource group.
+    plan: 
+        description:
+            - "It can be resource id of existing app service plan. eg.,
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_functionapp.py:39]: as a common practice, could you pls add support of resourcegroup+name/name also? it  can be a separate pr.

File: lib/ansible/modules/cloud/azure/azure_rm_functionapp.py [hunk 3]
```diff
@@ -218,7 +225,8 @@ def exec_module(self, **kwargs):
                     site_config=SiteConfig(
                         app_settings=self.aggregated_app_settings(),
                         scm_type='LocalGit'
-                    )
+                    ),
+                    server_farm_id=self.plan
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_functionapp.py:230]: check changed or not for check_mode?

Comment (yungezz) [reply lib/ansible/modules/cloud/azure/azure_rm_functionapp.py:230]: pls add test to cover the change

**Comment by Fred-sun** [comment]
@goettl79 Thanks for your contribution, Could you help changed according by yungezz's comment? Thanks!

File: lib/ansible/modules/cloud/azure/azure_rm_functionapp.py (hunk 2 above)
Comment (goettl79) [reply lib/ansible/modules/cloud/azure/azure_rm_functionapp.py:39]: Probably in a new pull request. Will leave it as it is in this one.

File: lib/ansible/modules/cloud/azure/azure_rm_functionapp.py (hunk 3 above)
Comment (goettl79) [reply lib/ansible/modules/cloud/azure/azure_rm_functionapp.py:230]: Hi, I could not find the location of existing tests. Could you point me to an existing unit test?

Comment (Fred-sun) [reply lib/ansible/modules/cloud/azure/azure_rm_functionapp.py:230]: @goettl79 Maybe you can try this [link-related](https://github.com/ansible/ansible/blob/devel/test/integration/targets/azure_rm_functionapp/tasks/main.yml) functionapp test, Thanks!

**Comment by Fred-sun** [comment]
kindly ping

**Comment by goettl79** [comment]
argh. The create function app plan functionality is missing to make an end to end integration test. I forgot that I did that manually. 

Currently, I have no time to implement that too. I'm closing this request.
